### PR TITLE
[SYCL] fix opaque pointer issue in compileTimeProperty alignment

### PR DESCRIPTION
**Pull request by broxigarchen**
update compileTimeProperties pass to check opaque pointers

### CODE REVIEW

File: llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp [hunk 1]
```diff
@@ -525,9 +525,10 @@ void CompileTimePropertiesPass::parseAlignmentAndApply(
   // Get the global variable with the annotation string.
   const GlobalVariable *AnnotStrArgGV = nullptr;
   const Value *IntrAnnotStringArg = IntrInst->getArgOperand(1);
-  if (auto *GEP = dyn_cast<GEPOperator>(IntrAnnotStringArg))
-    if (auto *C = dyn_cast<Constant>(GEP->getOperand(0)))
-      AnnotStrArgGV = dyn_cast<GlobalVariable>(C);
+  if (IntrAnnotStringArg->getType()->isOpaquePointerTy())
+    AnnotStrArgGV = dyn_cast<GlobalVariable>(IntrAnnotStringArg);
+  else if (auto *GEP = dyn_cast<GEPOperator>(IntrAnnotStringArg))
```
Comment (LU-JOHN) [inline llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp:530]: The handling for this case (which does not seem to be related to opaque pointers) has changed.  What's the rationale for this change?

Comment (broxigarchen) [reply llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp:530]: This is for the opaque pointer. If the arg on ptr.annotation is an opaque pointer we do things differently

**Review by LU-JOHN — APPROVED** [review]
LGTM

**Comment by LU-JOHN** [comment]
@intel/llvm-gatekeepers  Can this be merged?  It has a bug fix related to opaque pointers.
